<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cRegulome">
<title>Using cRegulome • cRegulome</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using cRegulome">
<meta property="og:description" content="cRegulome">
<meta property="og:image" content="https://docs.ropensci.org/cRegulome/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">cRegulome</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.3.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/case_study.html">Case Study</a>
    <a class="dropdown-item" href="../articles/using_cRegulome.html">Using cRegulome</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/cRegulome/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Using cRegulome</h1>
                        <h4 data-toc-skip class="author">Mahmoud Ahmed</h4>
            
            <h4 data-toc-skip class="date">August 22, 2017</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/cRegulome/blob/HEAD/vignettes/using_cRegulome.Rmd" class="external-link"><code>vignettes/using_cRegulome.Rmd</code></a></small>
      <div class="d-none name"><code>using_cRegulome.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Transcription factors and microRNAs are important for regulating the gene expression in normal physiology and pathological conditions. Many bioinformatic tools were built to predict and identify transcription factors and microRNA targets and their role in development of diseases including cancers. The availability of public access high-throughput data allowed for data-driven discoveries and validations of these predictions. Here, we build on that kind of tools and integrative analyses to provide a tool to access, manage and visualize data from open source databases. cRegulome provides a programmatic access to the regulome (microRNA and transcription factor) correlations with target genes in cancer. The package obtains a local instance of Cistrome Cancer and miRCancerdb databases and provides objects and methods to interact with and visualize the correlation data.</p>
</div>
<div class="section level2">
<h2 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>To get started with cRegulome, we show a very quick example. We first start by downloading a small test database file, make a simple query and convert the output to a cRegulome object to print and visualize.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load required libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/cRegulome">cRegulome</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rsqlite.r-dbi.org" class="external-link">RSQLite</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the db file when using it for the first time</span></span>
<span><span class="va">destfile</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">'cRegulome.db.gz'</span>, sep <span class="op">=</span> <span class="st">'/'</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">destfile</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/get_db.html">get_db</a></span><span class="op">(</span>test <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># connect to the db file</span></span>
<span><span class="va">db_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">'cRegulome.db'</span>, sep <span class="op">=</span> <span class="st">'/'</span><span class="op">)</span></span>
<span><span class="va">conn</span> <span class="op">&lt;-</span> <span class="fu">dbConnect</span><span class="op">(</span><span class="fu"><a href="https://rsqlite.r-dbi.org/reference/SQLite.html" class="external-link">SQLite</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">db_file</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># alternative to downloading the database file</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://s3-eu-west-1.amazonaws.com/pfigshare-u-files/9537385/cRegulome.db.gz</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> cRegulome.db.gz</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># enter a custom query with different arguments</span></span>
<span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mir.html">get_mir</a></span><span class="op">(</span><span class="va">conn</span>,</span>
<span>               mir <span class="op">=</span> <span class="st">'hsa-let-7g'</span>,</span>
<span>               study <span class="op">=</span> <span class="st">'STES'</span>,</span>
<span>               min_abs_cor <span class="op">=</span> <span class="fl">.3</span>,</span>
<span>               max_num <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># make a cmicroRNA object   </span></span>
<span><span class="va">ob</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cmicroRNA.html">cmicroRNA</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># print object</span></span>
<span><span class="va">ob</span></span></code></pre></div>
<pre><code><span><span class="co">## A cmicroRNA object: microRNA-gene correlations in Cancer</span></span>
<span><span class="co">##  Contains:</span></span>
<span><span class="co">##  1 Cancer study/ies: STES </span></span>
<span><span class="co">##  1 microRNA/s: hsa-let-7g </span></span>
<span><span class="co">##  5 features: ARHGEF15 LSM11 STARD13 CXorf36 IL13</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot object</span></span>
<span><span class="fu"><a href="../reference/cor_plot.html">cor_plot</a></span><span class="op">(</span><span class="va">ob</span><span class="op">)</span></span></code></pre></div>
<p><img src="using_cRegulome_files/figure-html/plot_object-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="package-description">Package Description<a class="anchor" aria-label="anchor" href="#package-description"></a>
</h2>
<div class="section level3">
<h3 id="data-sources">Data sources<a class="anchor" aria-label="anchor" href="#data-sources"></a>
</h3>
<p>The two main sources of data used by this package are Cistrome Cancer and miRCancerdb databases. Cistrome Cancer is based on an integrative analysis of The Cancer Genome Atlas (TCGA) and public ChIP-seq data. It provides calculated correlations of (n = 320) transcription factors and their target genes in (n = 29) cancer study. In addition, Cistrome Cancer provides the transcription factors regulatory potential to target and non-target genes. miRCancerdb uses TCGA data and TargetScan annotations to correlate known microRNAs (n = 750) and target and non-target genes in (n = 25) cancer studies.</p>
</div>
<div class="section level3">
<h3 id="database-file">Database file<a class="anchor" aria-label="anchor" href="#database-file"></a>
</h3>
<p>cRegulome obtains a pre-build SQLite database file of the Cistrome Cancer and miRCancerdb databases. The details of this build is provided at <a href="https://github.com/MahShaaban/cRegulomedb" class="external-link">cRegulomedb</a> in addition to the scripts used to pull, format and deposit the data at an on-line repository. Briefly, the SQLite database consist of 4 tables <code>cor_mir</code> and <code>cor_tf</code> for correlation values and <code>targets_mir</code> and <code>targets_tf</code> for microRNA miRBase ID and transcription factors symbols to genes mappings. Two indices were created to facilitate the database search using the miRBase IDs and transcription factors symbols. The database file can be downloaded using the function <code>get_db</code>.</p>
<p>To show the details of the database file, the following code connects to the database and show the names of tables and fields in each of them.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># table names</span></span>
<span><span class="va">tabs</span> <span class="op">&lt;-</span> <span class="fu">dbListTables</span><span class="op">(</span><span class="va">conn</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">tabs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "cor_mir"     "cor_tf"      "targets_mir" "targets_tf"</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># fields/columns in the tables</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">tabs</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">dbListFields</span><span class="op">(</span><span class="va">conn</span>, <span class="va">tabs</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "mirna_base" "feature"    "STES"      </span></span>
<span><span class="co">## [1] "tf"      "feature" "STES"   </span></span>
<span><span class="co">## [1] "mirna_base" "feature"   </span></span>
<span><span class="co">## [1] "tf"      "feature" "study"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="database-query">Database query<a class="anchor" aria-label="anchor" href="#database-query"></a>
</h3>
<p>To query the database using cRegulome, we provide two main functions; <code>get_mir</code> and <code>get_tf</code> for querying microRNA and transcription factors correlations respectively. Users need to provide the proper IDs for microRNA, transcription factor symbols and/or TCGA study identifiers. microRNAs are referred to by the official miRBase IDs, transcription factors by their corresponding official gene symbols that contains them and TCGA studies with their common identifiers. In either cases, the output of calling the these functions is a tidy data frame of 4 columns; <code>mirna_base</code>/ <code>tf</code>, <code>feature</code>, <code>cor</code> and <code>study</code> These correspond to the miRBase IDs or transcription factors symbol, gene symbol, correlation value and the TCGA study identifier.</p>
<p>Here we show an example of such a query. Then, we illustrate how this query is executed on the database using basic <code>RSQLite</code> and <code>dbplyr</code> which is what the <code>get_*</code> functions are doing.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># query the db for two microRNAs</span></span>
<span><span class="va">dat_mir</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mir.html">get_mir</a></span><span class="op">(</span><span class="va">conn</span>,</span>
<span>                   mir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'hsa-let-7g'</span>, <span class="st">'hsa-let-7i'</span><span class="op">)</span>,</span>
<span>                   study <span class="op">=</span> <span class="st">'STES'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># query the db for two transcription factors</span></span>
<span><span class="va">dat_tf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_tf.html">get_tf</a></span><span class="op">(</span><span class="va">conn</span>,</span>
<span>                 tf <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'LEF1'</span>, <span class="st">'MYB'</span><span class="op">)</span>,</span>
<span>                 study <span class="op">=</span> <span class="st">'STES'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show first 6 line of each of the data.frames</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dat_mir</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dat_tf</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   mirna_base feature   cor study</span></span>
<span><span class="co">## 1 hsa-let-7g    ABL2  0.11  STES</span></span>
<span><span class="co">## 2 hsa-let-7g  ACTR10 -0.16  STES</span></span>
<span><span class="co">## 3 hsa-let-7g  ACVR1C -0.13  STES</span></span>
<span><span class="co">## 4 hsa-let-7g ADAMTS1  0.27  STES</span></span>
<span><span class="co">## 5 hsa-let-7g ADAMTS5  0.20  STES</span></span>
<span><span class="co">## 6 hsa-let-7g   ADCY9  0.18  STES</span></span></code></pre>
<pre><code><span><span class="co">##     tf feature  cor study</span></span>
<span><span class="co">## 1 LEF1     A2M 0.55  STES</span></span>
<span><span class="co">## 2 LEF1  A4GALT 0.40  STES</span></span>
<span><span class="co">## 3 LEF1   ABCA1 0.44  STES</span></span>
<span><span class="co">## 4 LEF1   ABCA6 0.54  STES</span></span>
<span><span class="co">## 5 LEF1   ABCA8 0.41  STES</span></span>
<span><span class="co">## 6 LEF1   ABCA9 0.48  STES</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="objects">Objects<a class="anchor" aria-label="anchor" href="#objects"></a>
</h3>
<p>Two S3 objects are provided by cRegulome to store and dispatch methods on the correlation data. cmicroRNA and cTF for microRNA and transcription factors respectively. The structure of these objects is very similar. Basically, as all S3 objects, it’s a list of 4 items; microRNA or TF for the regulome element, features for the gene hits, studies for the TCGA studies and finally corr is either a <code>data.frame</code> when the object has data.from a single TCGA study or a named list of data.frames when it has data from multiple studies. Each of these data.frames has the regulome element (microRNAs or transcription factors) in columns and features/genes in rows.<br>
To construct these objects, users need to call a constructor function with the corresponding names on the data.frame output form <code>get_*</code>. The reverse is possible by calling the function <code>cor_tidy</code> on the object to get back the tidy data.frame.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># explore the cmicroRNA object</span></span>
<span><span class="va">ob_mir</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cmicroRNA.html">cmicroRNA</a></span><span class="op">(</span><span class="va">dat_mir</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">ob_mir</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "cmicroRNA"</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">ob_mir</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 4</span></span>
<span><span class="co">##  $ microRNA: chr [1:2] "hsa-let-7g" "hsa-let-7i"</span></span>
<span><span class="co">##  $ features: chr [1:612] "ABL2" "ACTR10" "ACVR1C" "ADAMTS1" ...</span></span>
<span><span class="co">##  $ studies : chr "STES"</span></span>
<span><span class="co">##  $ corr    :List of 1</span></span>
<span><span class="co">##   ..$ STES:'data.frame': 612 obs. of  2 variables:</span></span>
<span><span class="co">##   .. ..$ hsa-let-7g: num [1:612] NA NA NA NA 0.11 NA NA NA -0.16 NA ...</span></span>
<span><span class="co">##   .. ..$ hsa-let-7i: num [1:612] -0.2 0.12 0.14 -0.1 0.27 -0.12 0.13 0.1 NA 0.23 ...</span></span>
<span><span class="co">##  - attr(*, "class")= chr "cmicroRNA"</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># explore the cTF object</span></span>
<span><span class="va">ob_tf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cTF.html">cTF</a></span><span class="op">(</span><span class="va">dat_tf</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">ob_tf</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "cTF"</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">ob_tf</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 4</span></span>
<span><span class="co">##  $ TF      : chr [1:2] "LEF1" "MYB"</span></span>
<span><span class="co">##  $ features: chr [1:1953] "A2M" "A4GALT" "ABCA1" "ABCA6" ...</span></span>
<span><span class="co">##  $ studies : chr "STES"</span></span>
<span><span class="co">##  $ corr    :List of 1</span></span>
<span><span class="co">##   ..$ STES:'data.frame': 1953 obs. of  2 variables:</span></span>
<span><span class="co">##   .. ..$ LEF1: num [1:1953] 0.55 0.4 0.44 0.54 0.41 0.48 0.56 0.43 0.54 0.45 ...</span></span>
<span><span class="co">##   .. ..$ MYB : num [1:1953] NA NA NA NA NA NA NA NA NA NA ...</span></span>
<span><span class="co">##  - attr(*, "class")= chr "cTF"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a>
</h3>
<p>cRegulome provides S3 methods to interact a visualize the correlations data in the cmicroRNA and cTF objects. Table 1 provides an over view of these functions. These methods dispatch directly on the objects and could be customized and manipulated in the same way as their generics.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cmicroRNA object methods</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/methods.html" class="external-link">methods</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">'cmicroRNA'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] cor_hist         cor_igraph       cor_joy          cor_plot        </span></span>
<span><span class="co">## [5] cor_tidy         cor_upset        cor_venn_diagram print           </span></span>
<span><span class="co">## see '?methods' for accessing help and source code</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cTF object methods</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/methods.html" class="external-link">methods</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">'cTF'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] cor_hist         cor_igraph       cor_joy          cor_plot        </span></span>
<span><span class="co">## [5] cor_tidy         cor_upset        cor_venn_diagram print           </span></span>
<span><span class="co">## see '?methods' for accessing help and source code</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># tidy method</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/cor_tidy.html">cor_tidy</a></span><span class="op">(</span><span class="va">ob_mir</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    mirna_base feature   cor study</span></span>
<span><span class="co">## 5  hsa-let-7g    ABL2  0.11  STES</span></span>
<span><span class="co">## 9  hsa-let-7g  ACTR10 -0.16  STES</span></span>
<span><span class="co">## 11 hsa-let-7g  ACVR1C -0.13  STES</span></span>
<span><span class="co">## 13 hsa-let-7g ADAMTS1  0.27  STES</span></span>
<span><span class="co">## 16 hsa-let-7g ADAMTS5  0.20  STES</span></span>
<span><span class="co">## 17 hsa-let-7g   ADCY9  0.18  STES</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cor_hist method</span></span>
<span><span class="fu"><a href="../reference/cor_hist.html">cor_hist</a></span><span class="op">(</span><span class="va">ob_mir</span>,</span>
<span>     breaks <span class="op">=</span> <span class="fl">100</span>,</span>
<span>     main <span class="op">=</span> <span class="st">''</span>, xlab <span class="op">=</span> <span class="st">'Correlation'</span><span class="op">)</span></span></code></pre></div>
<p><img src="using_cRegulome_files/figure-html/cor_hist_method-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## null device </span></span>
<span><span class="co">##           1</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cor_joy method</span></span>
<span><span class="fu"><a href="../reference/cor_joy.html">cor_joy</a></span><span class="op">(</span><span class="va">ob_mir</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Correlation'</span>, y <span class="op">=</span> <span class="st">''</span><span class="op">)</span></span></code></pre></div>
<p><img src="using_cRegulome_files/figure-html/cor_joy_method-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## null device </span></span>
<span><span class="co">##           1</span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cor_venn_diagram method</span></span>
<span><span class="fu"><a href="../reference/cor_venn_diagram.html">cor_venn_diagram</a></span><span class="op">(</span><span class="va">ob_mir</span>, cat.default.pos <span class="op">=</span> <span class="st">'text'</span><span class="op">)</span></span></code></pre></div>
<p><img src="using_cRegulome_files/figure-html/cor_venn_diagram_method-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## null device </span></span>
<span><span class="co">##           1</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cor_upset method</span></span>
<span><span class="fu"><a href="../reference/cor_upset.html">cor_upset</a></span><span class="op">(</span><span class="va">ob_mir</span><span class="op">)</span></span></code></pre></div>
<p><img src="using_cRegulome_files/figure-html/cor_upset_method-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## null device </span></span>
<span><span class="co">##           1</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="contributions">Contributions<a class="anchor" aria-label="anchor" href="#contributions"></a>
</h2>
<p>Comments, issues and contributions are welcomed at: <a href="https://github.com/MahShaaban/cRegulome" class="external-link">https://github.com/MahShaaban/cRegulome</a></p>
</div>
<div class="section level2">
<h2 id="citations">Citations<a class="anchor" aria-label="anchor" href="#citations"></a>
</h2>
<p>Please cite:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">'cRegulome'</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
